### ISSUE TITLE
MacOS: closing the window quits the application

### ISSUE THREAD

**Issue by davecotter** [sqlitebrowser/sqlitebrowser#2673]
Details for the issue
--------------------
#### What did you do?
click the red dot

#### What did you expect to see?
the window close, so i can open a new one

#### What did you see instead?
the app quit

Useful extra information
-------------------------
on windows, this is kindof normal, but on mac it most definitely is not. 
i close a window because i want a fresh start, to open a new database
to quit an app, use the file menu->quit or command Q.  closing a window should not have a secondary effect of quitting the app, that is very surprising for mac users

> DB4S v3.12.1 [built for x86_64-little_endian-lp64] on macOS Mojave (10.14) (darwin/18.7.0) [x86_64]
> using SQLCipher Version 4.4.0 community (based on SQLite 3.31.0)
> and Qt 5.15.1

**Comment by chrisjlocke** [comment]
I've never used a Mac, so can't really comment - Justin is out resident Mac user.
As a Windows user, it's been ingrained that closing a window closes the application - even closing the last tab in Firefox closes the application.  Must admit closing a window to 'start again' does seem an odd workflow.  As DB4S is a 'single database' application, by opening a database, you are in fact 'starting again'.  

I'll tag this with 'enhancement', but this is the first issue I've seen regarding this phenomenon.  I guess we could add code in the close button that says, "Are we on a Mac?  If so, close the database instead" but really it should warrant some more love before something like that is added - some other users should also suggest this.

**Comment by davecotter** [comment]
right, but, see, as a windows user, you are unaware of what it's like to be a mac user.

as a mac user, you close a window, you open another one. all apps behave the same way. it's JARRING and SURPRISING to have the app "just disappear" only because you close a window, because, you see, on mac, the menu bar is NOT PART OF THE WINDOW.

so users expect to continue useing the menu bar after they close a window, for example to make a new window. 
it's been like this since 1984.  mac users going to windows are surprised by this, and windows users going to mac are surprised by this.

no need to take my word for it, as any long time mac user if this is expected behavior.

thanks.

**Comment by scottfurry** [comment]
He's right (speak as an occasional Mac user since Apple ][ days). Closing a window in current MacOS leaves the application resident unless fully terminated via menu command. Menu bar will will still show application is active (and/or cycling through cmd-tab to see current active applications). Expectation that closing a window on MacOS closes the application is not inline with the `Human Interface Guidelines`.

**Comment by justinclift** [comment]
Yeah, this is a good point.  We should probably figure out if there's a way to leave DB4S running (with active menu bar, etc) when the last window is closed.

DB4S would probably just need to be in a similar (or the same) state as when DB4S is first opened without any active database.  eg waiting for the user to "do something" in order to take action
